### PR TITLE
`create-if` may fail

### PR DESCRIPTION
**Pull request by jmpesp**
`create-if` may fail if the vnic being used inside a zone already has an interface in that zone. We saw this when bringing up dogfood, but I didn't see this in my original testing.

In `RunningZone::boot`, eat errors if the ipadm command is `create-if`

### CODE REVIEW

**Comment by jmpesp** [comment]
A few private messages from Andy lead me to the real fix for this: self-assembling zones will run `ipadm create-addr`, which will create IP interfaces, and conflict with calling `create-if` explicitly.

**Comment by jmpesp** [comment]
`helios / package` is failing with 

```
Error: Digest mismatch downloading dendrite-softnpu: Saw 5a36b855551f07ed4399497652b69be0619eaeccee6e67389f297af4599a2b51, expected 19fcb003b5070e0b8d029f015d2ba31a00518317bffa0fc24a58058e0ed5b32c
```

but

```
+ curl -sqL https://buildomat.eng.oxide.computer/public/file/oxidecomputer/dendrite/image/60a7c97d08d9fc26cda1e133f7f25e65895fe167/dendrite-softnpu.tar.gz
+ sha256sum -
19fcb003b5070e0b8d029f015d2ba31a00518317bffa0fc24a58058e0ed5b32c  -
```

**Comment by jmpesp** [comment]
the commit sha is from

```toml
[package.dendrite-softnpu]
service_name = "dendrite"
only_for_targets.switch = "softnpu"
only_for_targets.image = "standard"
# To manually override the package source:
#
# 1. Build the zone image manually
#   1a. cd <dendrite tree>
#   1b. cargo build --features=softnpu --release
#   1c. cargo xtask dist -o -r --features softnpu
# 2. Copy dendrite.tar.gz from dendrite/out to omicron/out/dendrite-softnpu.tar.gz
source.type = "prebuilt"
source.repo = "dendrite"
source.commit = "60a7c97d08d9fc26cda1e133f7f25e65895fe167"
source.sha256 = "19fcb003b5070e0b8d029f015d2ba31a00518317bffa0fc24a58058e0ed5b32c"
output.type = "zone"
output.intermediate_only = true
```

**Comment by jmpesp** [comment]
buildomat's sh256 agrees:

```
$ curl -sqL https://buildomat.eng.oxide.computer/public/file/oxidecomputer/dendrite/image/60a7c97d08d9fc26cda1e133f7f25e65895fe167/dendrite-softnpu.sha256.txt
```

**Comment by jmpesp** [comment]
If I run

```
	cargo run --locked --release --bin omicron-package -- -t test target create -i standard -m non-gimlet -s softnpu
	cargo run --locked --release --bin omicron-package -- -t test package
```

locally, I don't see the digest mismatch...?

**Review by andrewjstone — APPROVED** [review]
LGTM. Thanks for the fix!

File: illumos-utils/src/running_zone.rs [hunk 1]
```diff
@@ -234,7 +251,13 @@ impl RunningZone {
         ];
 
         for args in &commands {
-            running_zone.run_cmd(args)?;
+            if let Err(e) = running_zone.run_cmd(args) {
+                // `create-if` may fail if the vnic already has an interface.
+                // eat that error and continue.
+                if args[1] != "create-if" {
```
Comment (jgallagher) [inline illumos-utils/src/running_zone.rs:257]: This seems a little dicey in that it eats _any_ error from `create-if`. Any chance we could check that the error is specifically "this interface already exists"?

Comment (jmpesp) [reply illumos-utils/src/running_zone.rs]: Yeah, I realized after asking for a review that I left that bit in! See 676f760328b8fff79bc83b6f70b40f20040e9173 and 90cc69d2215debef8a475c65d93a2df98e3cf109

**Comment by jmpesp** [comment]
I'm deploying the latest to verify those changes, but thanks for the reviews!

File: illumos-utils/src/running_zone.rs (hunk 1 above)
Comment (jgallagher) [reply illumos-utils/src/running_zone.rs]: Nice, those changes LGTM. Thanks!

**Comment by jmpesp** [comment]
Deploy looks good, waiting on re-reviews.

**Review by jgallagher — APPROVED** [review]

**Pull request merged**